src/SFA.DAS.AssessorService.Web/Controllers/ApplyForWithdrawalController.cs

```diff
@@ -142,266 +132,82 @@ public async Task<IActionResult> ChooseStandardForWithdrawal(int? pageIndex)
+            var standardVersions = await _standardVersionApiClient.GetEpaoRegisteredStandardVersions(organisation.EndPointAssessorOrganisationId, model.IfateReferenceNumber);
+            var standard = standardVersions.FirstOrDefault();
 
             if (!ModelState.IsValid)
             {
-                model.IFateReferenceNumber = iFateReferenceNumber;
-                model.Level = standard?.Level ?? default;
-                model.StandardName = standard?.Title;
-                model.BackAction = backAction;
-                return View(model);
+                return RedirectToRoute(CheckWithdrawalRequestRouteGet, new { ifateReferenceNumber = model.IfateReferenceNumber });
             }
 
             if (model.Continue.Equals("no", StringComparison.InvariantCultureIgnoreCase))
-                return RedirectToAction(
-                        nameof(WithdrawalApplications),
-                        nameof(ApplyForWithdrawalController).RemoveController());
+                return RedirectToRoute(WithdrawalApplicationsRouteGet);
 
-            if (string.IsNullOrWhiteSpace(iFateReferenceNumber))
+            if (string.IsNullOrWhiteSpace(model.IfateReferenceNumber))
             {
                 var id = await CreateOrganisationWithdrawalApplication(contact, organisation);
 
-                return RedirectToAction(
-                    nameof(ApplicationController.Sequence),
-                    nameof(ApplicationController).RemoveController(),
+                return RedirectToRoute(ApplicationController.SequenceRouteGet,
                     new { Id = id, sequenceNo = ApplyConst.ORGANISATION_WITHDRAWAL_SEQUENCE_NO });
             }
             else
             {
-                if (string.IsNullOrWhiteSpace(versionsToWithdrawal))
-                {
-                    var applications = await GetInProgressWithdrawalVersionApplications(contact.Id, iFateReferenceNumber);
-                    if (applications.Any())
-                    {
-                        await _applicationApiClient.DeleteApplications(new DeleteApplicationsRequest()
-                        {
-                            ApplicationIds = applications.Select(x => x.Id),
-                            DeletingContactId = null
-                        });
-                    }
-                }
-
-                var id = await CreateWithdrawalApplication(contact, organisation,
+                var id = await CreateStandardWithdrawalApplication(contact, organisation,
                             standard.LarsCode,
-                            iFateReferenceNumber,
-                            standard.Title,
-                            string.IsNullOrWhiteSpace(versionsToWithdrawal) ? StandardOrVersion.Standard : StandardOrVersion.Version,
-                            versionsToWithdrawal?.Split(","));
+                            model.IfateReferenceNumber,
+                            standard.Title);
```

Comment:
Since `standard` could be null because it was assigned after a `FirstOrDefault()`, we should either account for the possibility of null, or use `First()` instead

Reply:
Will use first as this would be exception to have been passed an IfateReferenceNumber that the organisation did not assess